### PR TITLE
Trim readme, offloading to the manual

### PR DESCRIPTION
**Pull request by Mats-SX**
Removes most of the content from the README.
Some bits were moved to the manual, which is the thing that obsoleted the README content.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,284 +30,57 @@ pip install graphdatascience
...
-
-res = gds.shortestPath.dijkstra.stream(G, sourceNode=source_id, targetNode=target_id)
-assert res["totalCost"][0] == 100
-```
-
-The nodes found by `gds.find_node_id` are those that have all labels specified and fully match all property key-value pairs given.
-Note that exactly one node per method call must be matched.
-
-For more advanced filtering we recommend users do matching via Cypher's `MATCH`.
+* [Product recommendations with kNN based on FastRP embeddings](https://github.com/neo4j/graph-data-science-client/tree/main/examples/fastrp-and-knn.ipynb)
```
Comment (adamnsch) [inline README.md:81]: Noticing that this list is rather incomplete (not related to this PR)

File: doc/modules/ROOT/pages/pipelines.adoc [hunk 2]
```diff
@@ -4,10 +4,9 @@
 The Python client has special support for https://neo4j.com/docs/graph-data-science/current/machine-learning/linkprediction-pipelines/link-prediction/[Link prediction pipelines] and https://neo4j.com/docs/graph-data-science/current/machine-learning/node-property-prediction/[pipelines for node property prediction].
 The GDS pipelines are represented as pipeline objects.
 
-The Python method calls to _create_ the pipelines match their Cypher counterparts exactly.
-However, the rest of the pipeline functionality is deferred to methods on the pipeline objects themselves.
-Once created, the `TrainingPipeline` can be passed as arguments to methods in the Python client, such as the https://neo4j.com/docs/graph-data-science/current/pipeline-catalog/pipeline-catalog/[pipeline catalog operations].
-Additionally, the `TrainingPipeline` has convenience methods allowing for inspection of the pipeline represented without explicitly involving the pipeline catalog.
+While it is possible to operate pipelines using the same signatures as the Cypher procedures, the GDS Python Client offers a more convenient way by using the pipeline objects.
```
Comment (adamnsch) [inline doc/modules/ROOT/pages/pipelines.adoc:7]: This is not actually true I think. Only the create procedures are mapped to Python methods. Unless you mean by using `run_cypher`?

File: README.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ It enables users to write pure Python code to project graphs, run algorithms, as
 
 The API is designed to mimic the GDS Cypher procedure API in Python code.
 It abstracts the necessary operations of the [Neo4j Python driver](https://neo4j.com/docs/python-manual/current/) to offer a simpler surface.
+Additionally, the client-specific graph, model, and pipeline objects offer convenient functions that heavily reduce the need to use Cypher to access and operate these GDS resources.
```
Comment (adamnsch) [inline README.md:16]: Nice!

File: README.md [hunk 4]
```diff
@@ -29,284 +30,57 @@ pip install graphdatascience
 ```
```
Comment (adamnsch) [inline README.md:31]: I wonder if we should also have a `## Documentation` header under which we refer to the manual too, just to be abundantly clear?

**Review by adamnsch — APPROVED** [review]
Nice work! Happy this is finally happening :tada: Just have some minor questions

File: README.md (hunk 3 above)
Comment (adamnsch) [reply README.md:16]: I wonder if we should also add this addition to the front page of the client manual?

File: README.md (hunk 1 above)
Comment (Mats-SX) [reply README.md:81]: Ach, I should've double-checked it. I'll have a look and cross-reference.

File: doc/modules/ROOT/pages/pipelines.adoc (hunk 2 above)
Comment (Mats-SX) [reply doc/modules/ROOT/pages/pipelines.adoc:7]: No I meant by mimicking -- will change this sentence.

File: README.md (hunk 3 above)
Comment (Mats-SX) [reply README.md:16]: Sure, I can do that.

File: README.md (hunk 4 above)
Comment (Mats-SX) [reply README.md:31]: Good idea!

File: README.md (hunk 1 above)
Comment (adamnsch) [reply README.md:81]: Fixed in this PR https://github.com/neo4j/graph-data-science-client/pull/249

Comment (Mats-SX) [reply README.md]: done

**Comment by Mats-SX** [comment]
@adamnsch All comments resulted in an update. Thank you. I will merge this, but you may still want to check the additional changes.

**Comment by adamnsch** [comment]
Looks like there's a bunch of asciidoc errors:
```
asciidoctor: ERROR: pipelines.adoc: line 58: include file not found: /opt/teamcity-agent/work/4fe534e03e95c24/doc/modules/ROOT/pages/partial$/pipelines/shared-pipeline-methods.adoc
[12:54:50 ](https://live.neo4j-build.io/buildConfiguration/GraphAnalytics_GdsClientPullRequestsTest__GraphAnalytics_LibraryArtifacts_44/15309309?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandPull+Request+Details=true&expandBuildChangesSection=true&showLog=15309309_1329_1329&logFilter=debug&logView=linear)  asciidoctor: ERROR: pipelines.adoc: line 152: include file not found: /opt/teamcity-agent/work/4fe534e03e95c24/doc/modules/ROOT/pages/partial$/pipelines/shared-model-methods.adoc
[12:54:50 ](https://live.neo4j-build.io/buildConfiguration/GraphAnalytics_GdsClientPullRequestsTest__GraphAnalytics_LibraryArtifacts_44/15309309?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandPull+Request+Details=true&expandBuildChangesSection=true&showLog=15309309_1330_1330&logFilter=debug&logView=linear)  asciidoctor: ERROR: pipelines.adoc: line 237: include file not found: /opt/teamcity-agent/work/4fe534e03e95c24/doc/modules/ROOT/pages/partial$/pipelines/shared-pipeline-methods.adoc
[12:54:50 ](https://live.neo4j-build.io/buildConfiguration/GraphAnalytics_GdsClientPullRequestsTest__GraphAnalytics_LibraryArtifacts_44/15309309?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandPull+Request+Details=true&expandBuildChangesSection=true&showLog=15309309_1331_1331&logFilter=debug&logView=linear)  asciidoctor: ERROR: pipelines.adoc: line 333: include file not found: /opt/teamcity-agent/work/4fe534e03e95c24/doc/modules/ROOT/pages/partial$/pipelines/shared-model-methods.adoc
[12:54:50 ](https://live.neo4j-build.io/buildConfiguration/GraphAnalytics_GdsClientPullRequestsTest__GraphAnalytics_LibraryArtifacts_44/15309309?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandPull+Request+Details=true&expandBuildChangesSection=true&showLog=15309309_1332_1332&logFilter=debug&logView=linear)  asciidoctor: ERROR: pipelines.adoc: line 406: include file not found: /opt/teamcity-agent/work/4fe534e03e95c24/doc/modules/ROOT/pages/partial$/pipelines/shared-pipeline-methods.adoc
[12:54:50 ](https://live.neo4j-build.io/buildConfiguration/GraphAnalytics_GdsClientPullRequestsTest__GraphAnalytics_LibraryArtifacts_44/15309309?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandPull+Request+Details=true&expandBuildChangesSection=true&showLog=15309309_1333_1333&logFilter=debug&logView=linear)  asciidoctor: ERROR: pipelines.adoc: line 499: include file not found: /opt/teamcity-agent/work/4fe534e03e95c24/doc/modules/ROOT/pages/partial$/pipelines/shared-model-methods.adoc
```
I took the liberty of turning of auto-merge. But perhaps it's not related to this PR?

**Pull request merged**